src/Models/Config.php

```diff
@@ -3,44 +3,123 @@
 namespace Rapidez\Core\Models;
 
 use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Arr;
 use Illuminate\Support\Facades\Cache;
 use Rapidez\Core\Exceptions\DecryptionException;
+use Rapidez\Core\Facades\Rapidez;
+
+enum ConfigScopes
+{
+    case SCOPE_STORE;
+    case SCOPE_WEBSITE;
+    case SCOPE_DEFAULT;
+}
 
 class Config extends Model
 {
     protected $table = 'core_config_data';
 
     protected $primaryKey = 'config_id';
 
+    protected $configCache = [];
+
```

Comment:
We're not using this anymore right?

Reply:
That's right, it's only using the cache and local variables now.